tsdb/engine/tsm1/file_store.go

```diff
@@ -1118,6 +1130,11 @@ func (p *purger) purge() {
 			p.mu.Lock()
 			for k, v := range p.files {
 				if !v.InUse() {
+					// Remove any mmap references held by the index.
+					if p.fileStore.dereferencer != nil {
+						v.deref(p.fileStore.dereferencer)
```

Comment:
I added it to `FileStore.Replace` in 6614e0b but `FileStore.Remove` doesn't seem to actually close the file so it doesn't seem like the right place for dereference. Is it the purger that actually closes those files?